### ISSUE TITLE
Should setup_postdata trigger an error for overriding WP global?

### ISSUE THREAD

**Issue by grappler** [WordPress-Coding-Standards/WordPress-Coding-Standards#842]
While preparing to merge the WP Globals check for the `WordPress-Theme` ruleset I found a place when `setup_postdata()` is used an error is triggered.

Should the following code trigger an error? 

```php
global $post;
$post = get_post( '10' );
setup_postdata( $post );
wp_reset_postdata();
```

This is also used in Twenty Seventeen https://github.com/WordPress/WordPress/blob/master/wp-content/themes/twentyseventeen/inc/template-tags.php#L142

**Comment by westonruter** [comment]
Good point. Yeah, the `$post` global variable unfortunately tends to be overwritten.

**Comment by jrfnl** [comment]
Duplicate of #34 ?

**Comment by danyj** [comment]
@westonruter Is there an actual solution to this? We are overriding the Woo templates within the theme https://github.com/woocommerce/woocommerce/blob/master/templates/cart/cross-sells.php#L35 as previously asked here https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/913  and not sure what to do in ref to the sniffer error msg. 

Any help is appreciated.

**Comment by zanematthew** [comment]
Any traction on this?

It seems the solution is;
1. Update the [Official documentation for setup_postdata()](https://developer.wordpress.org/reference/functions/setup_postdata/), noting it is **best not to use the function**, due to it not being valid code based on WordPress' coding standards
2. Update existing custom code to just iterate over items as usual, and replace functions like `the_title()` with functions like `echo esc_html( get_the_title( $some_id ) )`

...anyway you get the idea.
